src/codeu/chat/client/commandline/Chat.java

```diff
@@ -570,18 +574,24 @@ public void invoke(Tokenizer args) {
       @Override
       public void invoke(Tokenizer args) {
         System.out.println("--- start of conversation ---");
-        for (MessageContext message = conversation.firstMessage();
-                            message != null;
-                            message = message.next()) {
-          System.out.println();
-          System.out.format("USER : %s\n", message.message.author);
-          System.out.format("SENT : %s\n", message.message.creation);
-          System.out.println();
-          System.out.println(message.message.content);
-          System.out.println();
-        }
-        System.out.println("---  end of conversation  ---");
-      }
+
+        try {
+          for (MessageContext message = conversation.firstMessage();
+                              message != null;
+                              message = message.next()) {
+            System.out.println();
+            System.out.format("USER : %s\n", message.message.author);
+            System.out.format("SENT : %s\n", message.message.creation);
+            System.out.println();
+            System.out.println(message.message.content);
+            System.out.println();
+          }
+        }
+        catch (Exception e) {
+          System.out.println("You do not have permission to view this content");
```

Comment:
Another downside is the contents of the error message should come from the server.